kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -524,7 +524,7 @@ protected void handleTopicMetadataRequest(KafkaHeaderAndRequest metadataHar,
         List<TopicMetadata> allTopicMetadata = Collections.synchronizedList(Lists.newArrayList());
         List<Node> allNodes = Collections.synchronizedList(Lists.newArrayList());
         // Get all kop brokers in local cache
-        allNodes.addAll(adminManager.getBrokers());
+        allNodes.addAll(adminManager.getBrokers(metadataHar.getListenerName()));
```

Comment:
Could we just use

```java
        allNodes.addAll(adminManager.getBrokers(advertisedEndPoint.getListenerName()));
```

here? Then we don't need to add a `listenerName` field to `KafkaHeaderAndRequest`.